riscv/mmu.h

```diff
@@ -275,21 +275,21 @@ class mmu_t
   inline icache_entry_t* refill_icache(reg_t addr, icache_entry_t* entry)
   {
     auto tlb_entry = translate_insn_addr(addr);
-    insn_bits_t insn = from_le(*(uint16_t*)(tlb_entry.host_offset + addr));
+    insn_bits_t insn = from_target(*(target_endian<uint16_t>*)(tlb_entry.host_offset + addr));
```

Comment:
This seems to contradict https://github.com/riscv/riscv-isa-manual/blob/master/src/intro.tex#L638-L646

Reply:
Interresting.  When was this added?  I was following the RISC-V Instruction Set Manual v2.2 which says that parcels are stored "in a memory halfword according to the implementation's natural endianness".

If the encoding has changed so that the parcels are always little endian internally as well, then I'll need to update the toolchain and the FPGA implementation too.  Should be a pretty small change though.

Reply:
git has a nice feature for that: https://github.com/riscv/riscv-isa-manual/blame/master/src/intro.tex#L638-L646

(Jun 21, 2019)

You may also be interested in https://github.com/riscv/riscv-isa-manual/blob/master/src/machine.tex#L722-L808

Reply:
Thanks for the `mstatus` pointer.  I did have bi-endian on my roadmap so it's nice to see that there is an official spec for it.  Right now I need to fix the toolchain and update this PR though.  :smile_cat: